File: activity/20200805/junhaolin/728.Self_Dividing_Numbers.cpp
class Solution {
public:
    vector<int> selfDividingNumbers(int left, int right) {
        vector<int> result;
        for (int i=left; i<=right; i++) {
            int n = i;
            while (n) {
                if (n % 10 == 0)
                    break;
                else if (i % (n%10) != 0)
                    break;
                
                n /= 10;
            }
            
            if(n == 0)
                result.push_back(i);
        }
        return result;
    }
};

File: activity/20200805/ninezero/728_self_dividing_numbers.go
func selfDividingNumbers(left int, right int) []int {
	var res []int
	for i := left; i <= right; i++ {
		if isSelfDividingNumber(i) {
			res = append(res, i)
		}
	}
	return res
}

func isSelfDividingNumber(num int) bool {
	length := len(strconv.Itoa(num))
	temp := num
	for i := 0; i < length; i++ {
		// 每一位数
		n := num % 10
		if n == 0 || (n != 0 && temp % n != 0) {
			return false
		}
		num /= 10
	}
	return true
}

